lib/index.js

```diff
@@ -48,7 +48,7 @@ module.exports = async function faucetDispatch(referenceDir, config,
 	if(serve) {
 		server.static(serve, assetManager.manifest.webRoot);
 	} else if(liveserve) {
-		server.live(liveserve, assetManager.manifest.webRoot);
+		server.live(liveserve, assetManager.manifest.webRoot, config.middleware);
```

Comment:
IIRC we'd previously opted to relegate dev-server config to the CLI, i.e. consciously keeping it out of the config file? (Thus the host/port parsing of the respective option.)

That's why I'm _slightly_ uncomfortable with a top-level `middleware` entry (also because it's exclusive to live-server). It might make sense to go with a more explicit `devServer: { host, port, middleware }` namespace now?

Reply:
There is a significant difference between host/port and middleware:

1. The middleware is something that all people share. Let's say we need to set a specific HTTP header (which is the reason for this PR), then this is shared between all people.
2. host/port can be different between people and use cases.

I think that the configuration inf faucet.config.js should be "shared", while the CLI can be individual adjustments.

I agree though that this should probably not be top level.